andersoncd/penalties.py

```diff
@@ -146,3 +146,41 @@ def subdiff_distance(self, w, grad, ws):
 
     def is_penalized(self, n_features):
         return self.weights != 0
+
+
+spec_MCP = [
+    ('alpha', float64),
+    ('gamma', float64),
+]
+
+
+@jitclass(spec_L1)
+class MCP(Penalty):
+    def __init__(self, alpha, gamma):
+        self.alpha = alpha
+        self.gamma = gamma
+
+    def value(self, w):
+        return MC_penalty(w, self.alpha, self.gamma)
+
+    def prox_1d(self, value, stepsize, j):
+        return FT(value, self.alpha * stepsize, self.gamma / stepsize)
```

Comment:
FT stands for firm thresholding ?
anyways, same, can you code the prox here directly and remove the FT function ?